### PR TITLE
chore: fix various typos and update to US English

### PR DESCRIPTION
**Pull request by adeyinkaezra123**
### 🔗 Linked issue
No linked issue(s)

### ❓ Type of change
- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)

### 📚 Description

Rectified typos and spelling mistakes.

### CODE REVIEW

File: docs/1.getting-started/12.upgrade.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ Static sites             | ✅             | ✅                | ✅
 The migration guide provides a step-by-step comparison of Nuxt 2 features to Nuxt 3 features and guidance to adapt your current application.
 
 ::alert{type=info}
-👉 Checkout the [**migration guide from From Nuxt 2 to Nuxt 3**](/docs/migration/overview).
+👉 Checkout the [**migration guide from Nuxt 2 to Nuxt 3**](/docs/migration/overview).
```
Comment (danielroe) [inline docs/1.getting-started/12.upgrade.md:38]:
```suggestion
👉 Check out the [**guide to migrating from Nuxt 2 to Nuxt 3**](/docs/migration/overview).
```

File: docs/2.guide/2.directory-structure/1.components.md [hunk 2]
```diff
@@ -344,7 +344,7 @@ In this case, the `.server` + `.client` components are two 'halves' of a compone
 ```html [pages/example.vue]
 <template>
   <div>
-    <!-- this component will render Comments.server server-side then Comments.client once mounted in client-side -->
+    <!-- this component will render Comments.server on the server-side then Comments.client once mounted on the client-side -->
```
Comment (danielroe) [inline docs/2.guide/2.directory-structure/1.components.md:347]:
```suggestion
    <!-- this component will render Comments.server on the server then Comments.client once mounted in the browser -->
```

**Review by danielroe — APPROVED** [review]
Thank you! ❤️

File: packages/kit/src/build.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ export interface ExtendConfigOptions {
    */
   client?: boolean
   /**
-   * Prepends the plugin to the array with `unshit()` instead of `push()`.
```
Comment (Lehoczky) [inline packages/kit/src/build.ts:31]: lol :D

Comment (adeyinkaezra123) [reply packages/kit/src/build.ts:31]: Yeah!, no shit!! 😂

File: test/basic.test.ts [hunk 4]
```diff
@@ -1896,8 +1896,8 @@ describe.skipIf(isDev() || isWindows || !isRenderingJson)('payload rendering', (
   it.skipIf(!isRenderingJson)('should not include server-component HTML in payload', async () => {
     const payload = await $fetch('/prefetch/server-components/_payload.json', { responseType: 'text' })
     const entries = Object.entries(parsePayload(payload))
-    const [key, serialisedComponent] = entries.find(([key]) => key.startsWith('AsyncServerComponent')) || []
-    expect(serialisedComponent).toEqual(key)
+    const [key, serializedComponent] = entries.find(([key]) => key.startsWith('AsyncServerComponent')) || []
+    expect(serializedComponent).toEqual(key)
```
[the remark this answers is not in the document]
Comment (adeyinkaezra123) [reply test/basic.test.ts:1900]: Oh, okay 👍

**Pull request merged**